lib/pbench/agent/tool_meister.py

```diff
@@ -504,24 +571,9 @@ def wait(self):
         message every 5 seconds in between.
         """
         if self.process is None:
-            self.logger.error("Nothing to terminate")
+            self.logger.error("No process for which to wait")
             return
-
-        sts = self._wait_for_process(self.process)
-        if sts is None:
-            self.process.kill()
-            self.logger.error(
-                "Killed un-responsive persistent tool %s after 30 seconds", self.name
-            )
-        elif sts != 0 and sts != -(signal.SIGTERM):
-            self.logger.warning(
-                "Persistent tool %s failed to return success, %d",
-                self.name,
-                self.process.returncode,
-            )
-        else:
-            self.logger.info("Stopped persistent tool %s", self.name)
-        self.process = None
+        self.process = self._wait_for_process_w_kill(self.process)
```

Comment:
I think this call is missing an argument.

Also, the last paragraph should be removed from the docstring for this function -- this function doesn't call `wait()`, it doesn't emit periodic log messages, and it doesn't know that the interval will be 30 seconds.

Reply:
Argument is deliberately omitted.  Will fix the comment.

Reply:
Unlike `_wait_for_process()`, `_wait_for_process_w_kill()` doesn't supply a default value for the second (explicit) argument (at least in the sources presented here)...so I would expect Python to report an error if this line gets executed (and I think LGTM agrees...).

So, sure, a comment might be nice, but, if you're not going to supply the last argument here, then you also need to fix the function definition....  🙂